winml/adapter/WinMLAdapterFactory.h

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+#ifdef __cplusplus
+#include <WinMLAdapter.h>
+using namespace Windows::AI::MachineLearning::Adapter;
+
+#else
+struct IWinMLAdapter;
+typedef struct IWinMLAdapter IWinMLAdapter;
+#endif
+
+#ifdef __cplusplus
```

Comment:
This can just be
STDAPI OrtGetWinMLAdapter(IWinMLAdapter** adapter); 

No need for ifdefs...